models/projection/powder/mantid_projection_calculator.py

```diff
@@ -33,7 +40,13 @@ def calculate_projection(self, input_workspace, axis1, axis2):
                 str(dim0.getMaximum()) + ',' + str(dim0.getNBins())
             dim1 = dim1.getName() + ',' + str(dim1.getMinimum()) + ',' +\
                 str(dim1.getMaximum()) + ',' + str(dim1.getNBins())
-            return SliceMD(InputWorkspace=output_workspace, OutputWorkspace=output_workspace, AlignedDim0=dim0,
-                           AlignedDim1=dim1)
+            SliceMD(InputWorkspace=output_workspace, OutputWorkspace=output_workspace, AlignedDim0=dim0,
+                    AlignedDim1=dim1)
         else:
             raise NotImplementedError("Not implemented axis1 = %s and axis2 = %s" % (axis1, axis2))
+        # Now scale the energy axis if required - ConvertToMD always gives DeltaE in meV
+        if units == WAVENUMBER_LABEL:
+            TransformMD(InputWorkspace=output_workspace, OutputWorkspace=output_workspace, Scaling=scale)
+            self._workspace_provider.get_workspace_handle(output_workspace).setComment('MSlice_in_wavenumber')
+        elif units != MEV_LABEL:
+            raise NotImplementedError("Unit %s not recognised. Only 'meV' and 'cm-1' implemented." % (units))
```

Comment:
Is this method missing a `return` statement. The previous version returned from the `ConvertToMD`/`SliceMD` call but nothing is returned here.

I thinks this only affects the `cli`. If I run
```python

import cli
w1 = cli.Load('MAR21335_Ei60.00meV.nxs')
w2 = cli.get_projection(w1, '|Q|', 'DeltaE')
```

I get an error:
```---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-db82bf2cbb25> in <module>()
----> 1 w2 = cli.get_projection(w1, '|Q|', 'DeltaE')

/mnt/data1/source/github/mantidproject/mslice/cli/_mslice_commmands.py in get_projection(input_workspace, axis1, axis2, units)
     83     if len(names) > 1:
     84         raise Exception('Too many left hand side arguments, %s' % str(names))
---> 85     RenameWorkspace(InputWorkspace=output_workspace, OutputWorkspace=names[0])
     86     return output_workspace
     87 

/opt/mantidnightly/bin/mantid/simpleapi.pyc in RenameWorkspace(InputWorkspace,[OutputWorkspace],[True||False], ****kwargs)
    490     _set_logging_option(algm, kwargs)
    491     for key, val in arguments.items():
--> 492         algm.setProperty(key, val)
    493 
    494     algm.execute()

ValueError: When converting parameter "InputWorkspace": Unable to extract boost::shared_ptr<Workspace> from Python object
```

The GUI operations are fine though.